modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,480 @@
+		/// Next burn event ID
+		NextBurnEventId get(fn next_burn_event_id): u32;
+		/// Registry of all active tokens
+		RenTokenRegistry get(fn ren_token_registry): map hasher(blake2_128_concat) TokenIdOf<T> => Option<RenTokenInfoType<T>>;
+	}
+}
+
+decl_event!(
+	pub enum Event<T> where
+		<T as frame_system::Config>::AccountId,
+		TokenId = TokenIdOf<T>,
+		Balance = BalanceOf<T>
+	{
+		/// Event trigger when token is added to the Registry
+		RenTokenAdded(TokenId),
+		/// Event trigger when token is updates in the Registry
+		RenTokenUpdated(TokenId),
+		/// Event trigger when token is deleted from the Registry
+		RenTokenDeleted(TokenId),
+		/// Event trigger when token is spent from the pallet account
+		RenTokenSpent(TokenId, AccountId, Balance),
+		/// Event trigger when token is minted to a user account
+		RenTokenMinted(TokenId, AccountId, Balance),
+		/// Event trigger when token is burnt from a user account
+		RenTokenBurnt(TokenId, AccountId, DestAddress, Balance),
+
+	}
+);
+
+decl_error! {
+	pub enum Error for Module<T: Config> {
+		/// The mint signature is invalid.
+		InvalidMintSignature,
+		/// Burn ID overflow.
+		BurnIdOverflow,
+		/// RenTokenAlready Exists
+		RenTokenAlreadyExists,
+		/// No token with this ren_token_id found
+		RenTokenNotFound,
+		/// Asset for the token does not exist
+		RenTokenAssetNotFound,
+		/// Token name exceeds length limit
+		RenTokenNameLengthLimitExceeded,
+		/// The to address provided to the burn function failed validation corresponding to the token's name
+		InvalidBurnToAddress,
+		/// An operation that was not expected to fail failed
+		UnexpectedError,
+		/// Minting this token has been disabled
+		RenTokenMintDisabled,
+		/// Burning this token has been disabled
+		RenTokenBurnDisabled
+
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Config> for enum Call where origin: T::Origin {
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		/// Instantiates the token in the Registry AFTER instantiating the asset in Assets
+		/// Requries max_zombies and min_balance as inputs to Assets
+		/// Max length of the token name implemented
+		#[weight = T::WeightInfo::add_ren_token()]
+		pub fn add_ren_token(
+			origin,
+			#[compact] _ren_token_id: TokenIdOf<T>,
+			_ren_token_name: Vec<u8>,
+			_ren_token_renvm_id: [u8; 32],
+			_ren_token_pub_key: [u8; 20],
+			_ren_token_mint_enabled: bool,
+			_ren_token_burn_enabled: bool,
+			_ren_token_mint_fee: u32,
+			_ren_token_burn_fee: u32,
+			_ren_token_max_zombies: u32,
+			_ren_token_min_balance: BalanceOf<T>,
+		) -> DispatchResult
+		{
+			T::ControllerOrigin::ensure_origin(origin)?;
+
+			ensure!(_ren_token_name.len()<=NAME_MAX_LENGTH.into(), Error::<T>::RenTokenNameLengthLimitExceeded);
+			ensure!(!<RenTokenRegistry<T>>::contains_key(&_ren_token_id), Error::<T>::RenTokenAlreadyExists);
+
+			T::Assets::force_create_asset(_ren_token_id, Self::account_id(), _ren_token_max_zombies, _ren_token_min_balance)?;
```

Comment:
This needs to be thought out a little more. If the root origin is required to create asset. Then there's going to be a whole bunch of delay getting new assets off the ground, though we are reducing the governance times. Ultimately though, this doesn't provide flexibility for the RenVM team to create new asses.

Reply:
I think we should talk to the Ren team more to figure out how they plan to utilise this and potentially make all admin functions `ensure_none(origin)?;` Or get a clear answer if the root origin should be used as the controller or if they'll even provide a substrate address upon setup. This way they can create new assets for this bridge.

Reply:
Yes, this needs to be discussed with RenVM.